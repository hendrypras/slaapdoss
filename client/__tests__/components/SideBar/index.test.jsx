/* eslint-disable react/prop-types */
import { render, fireEvent } from '@utils/testHelper';
import SideBar from '@components/SideBar';
import { MemoryRouter } from 'react-router-dom';

const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
  Link: ({ to, children }) => <a href={to}>{children}</a>,
}));
let wrapper;
const mockProps = {
  userProfile: { image_url: 'http://asdasdas', username: 'username' },
  locale: 'id',
};
beforeEach(() => {
  wrapper = render(
    <MemoryRouter>
      <SideBar userProfile={mockProps.userProfile} locale={mockProps.locale}>
        <div>children</div>
      </SideBar>
    </MemoryRouter>
  );
});

describe('SideBar', () => {
  test('Should render correctly', () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('sidebar')).toBeInTheDocument();
  });
  test('Should call navigate when button clicked', () => {
    const { getByTestId } = wrapper;
    const button = getByTestId('navigate-back-to-home');
    fireEvent.click(button);
    expect(mockNavigate).toHaveBeenCalledWith(`/`);
  });
  test('should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
