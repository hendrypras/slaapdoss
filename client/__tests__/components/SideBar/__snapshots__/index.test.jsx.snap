// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapper"
        data-testid="sidebar"
      >
        <aside
          class="aside"
        >
          <button
            class="wrapperLogo"
            data-testid="navigate-back-to-home"
            type="button"
          >
            <img
              alt="logo"
              class="logo"
              src="/logo.svg"
            />
            <div
              class="logoTitle"
            >
              slaapdoss
            </div>
          </button>
          <ul
            class="links"
          >
            <li
              class="link"
            >
              <a
                href="/dashboard/cabins"
              >
                Daftar Cabin
              </a>
            </li>
            <li
              class="link"
            >
              <a
                href="/dashboard/create-cabin"
              >
                Tambah Cabin
              </a>
            </li>
            <li
              class="link"
            >
              <a
                href="/dashboard/create-type-room"
              >
                Tambah Tipe Ruangan
              </a>
            </li>
            <li
              class="link"
            >
              <a
                href="/dashboard/create-room"
              >
                Tambah Ruangan
              </a>
            </li>
            <li
              class="link"
            >
              <a
                href="/dashboard/create-banner"
              >
                Tambah Banner
              </a>
            </li>
            <li
              class="link"
            >
              <a
                href="/dashboard/list-banner"
              >
                Daftar Banner
              </a>
            </li>
            <li
              class="link"
            >
              <a
                href="/dashboard/list-order"
              >
                Daftar Pesanan
              </a>
            </li>
          </ul>
          <div
            class="wrapperFooter"
          >
            <button
              class="btnAvatar"
              type="button"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
                  data-testid="PersonIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                  />
                </svg>
              </div>
            </button>
            <button
              class="wrapper logout"
              data-testid="button"
              type="button"
            >
              Keluar
            </button>
          </div>
        </aside>
        <div>
          <div
            class="wrapperBtnLang"
          >
            <button
              class="toggle"
              data-testid="button-lang"
              type="button"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular avatar css-1wlk0hk-MuiAvatar-root"
              >
                <img
                  class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                  src="/id.png"
                />
              </div>
              <div
                class="lang"
              >
                id
              </div>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium arrowIcon css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowRightOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                />
              </svg>
            </button>
          </div>
          <main
            class="main"
          >
            <div>
              children
            </div>
          </main>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper"
      data-testid="sidebar"
    >
      <aside
        class="aside"
      >
        <button
          class="wrapperLogo"
          data-testid="navigate-back-to-home"
          type="button"
        >
          <img
            alt="logo"
            class="logo"
            src="/logo.svg"
          />
          <div
            class="logoTitle"
          >
            slaapdoss
          </div>
        </button>
        <ul
          class="links"
        >
          <li
            class="link"
          >
            <a
              href="/dashboard/cabins"
            >
              Daftar Cabin
            </a>
          </li>
          <li
            class="link"
          >
            <a
              href="/dashboard/create-cabin"
            >
              Tambah Cabin
            </a>
          </li>
          <li
            class="link"
          >
            <a
              href="/dashboard/create-type-room"
            >
              Tambah Tipe Ruangan
            </a>
          </li>
          <li
            class="link"
          >
            <a
              href="/dashboard/create-room"
            >
              Tambah Ruangan
            </a>
          </li>
          <li
            class="link"
          >
            <a
              href="/dashboard/create-banner"
            >
              Tambah Banner
            </a>
          </li>
          <li
            class="link"
          >
            <a
              href="/dashboard/list-banner"
            >
              Daftar Banner
            </a>
          </li>
          <li
            class="link"
          >
            <a
              href="/dashboard/list-order"
            >
              Daftar Pesanan
            </a>
          </li>
        </ul>
        <div
          class="wrapperFooter"
        >
          <button
            class="btnAvatar"
            type="button"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
                data-testid="PersonIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
          </button>
          <button
            class="wrapper logout"
            data-testid="button"
            type="button"
          >
            Keluar
          </button>
        </div>
      </aside>
      <div>
        <div
          class="wrapperBtnLang"
        >
          <button
            class="toggle"
            data-testid="button-lang"
            type="button"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular avatar css-1wlk0hk-MuiAvatar-root"
            >
              <img
                class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="/id.png"
              />
            </div>
            <div
              class="lang"
            >
              id
            </div>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium arrowIcon css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowRightOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
              />
            </svg>
          </button>
        </div>
        <main
          class="main"
        >
          <div>
            children
          </div>
        </main>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
